app/views/ads/_ad.html.erb

```diff
@@ -1,61 +1,73 @@
-<div class="card mb-5 <%= ad.kind %>">
-  <div class="card-content">
-    <div class="content">
-      <div class="field is-grouped is-grouped-multiline">
-        <div class="control">
-          <div class="tags has-addons">
-            <%= category_tag ad %>
-          </div>
-        </div>
+<div class="tile is-parent mb-2 is-12">
+  <div class="tile is-child card is-flex is-flex-direction-column">
+    <div class="card-content" data-controller="card-dropdown">
 
-        <div class="control">
-          <div class="tags has-addons">
-            <span class="tag">Grad</span>
-            <span class="tag is-info"><%= ad.zip_and_city%></span>
+      <div class="is-flex is-justify-content-space-between mb-4">
+        <%= category_tag ad %>
+
+        <div class="dropdown is-right ml-2" data-card-dropdown-target="dropdown">
+          <div class="dropdown-trigger">
+            <button class="button" aria-haspopup="true" aria-controls="dropdown-menu-card-actions" data-action="card-dropdown#toggle">
+              <span class="icon">
+                <%= inline_svg_tag('dots-vertical.svg', aria: true) %>
+              </span>
+            </button>
+          </div>
+          <div class="dropdown-menu" id="dropdown-menu-card-actions" role="menu">
+            <div class="dropdown-content" data-controller="clipboard">
+              <a href="<%= ad_path(ad.to_param) %>" class="dropdown-item is-flex is-align-items-center">
+                <span class="icon is-small mr-2">
+                  <%= inline_svg_tag('link.svg', aria: true) %>
+                </span>
+                Otvori oglas
+              </a>
+              <a class="dropdown-item is-flex is-align-items-center" data-clipboard-target="button" data-clipboard-text="<%= ad_url(ad.to_param) %>">
+                <span class="icon is-small mr-2">
+                  <%= inline_svg_tag('clipboard.svg', aria: true) %>
+                </span>
+                Kopiraj poveznicu
+              </a>
+            </div>
           </div>
         </div>
+      </div>
+
+      <div class="content">
+        <h3 class="is-size-5"><%= ad.zip_and_city %></h3>
```

Comment:
🤔 I think we could even omit the zip code here. It can be added below with the rest of the address.